scripts/flights2.py

```diff
@@ -0,0 +1,727 @@
+# dependencies = [
+#     "niquests",
+#     "polars",
+# ]
+# ///
+from __future__ import annotations
+
+import asyncio
+import datetime as dt
+import gzip
+import io
+import logging
+import tomllib
+import zipfile
+from collections import defaultdict, deque
+from collections.abc import Iterable, Sequence
+from enum import StrEnum
+from functools import cached_property
+from pathlib import Path
+from typing import IO, TYPE_CHECKING
+
+import niquests
+import polars as pl
+from polars import col
+from polars import selectors as cs
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator, Mapping
+    from typing import Any, ClassVar, Literal, LiteralString
+
+
+logger = logging.getLogger(__name__)
+
+
+type Rows = Literal[
+    1_000,
+    2_000,
+    5_000,
+    10_000,
+    20_000,
+    100_000,
+    200_000,
+    500_000,
+    1_000_000,
+    3_000_000,
+    5_000_000,
+    10_000_000,
+    100_000_000,
+    500_000_000,
+    1_000_000_000,
+]
+"""Number of rows to include in the output."""
+
+type Extension = Literal[".arrow", ".csv", ".json", ".parquet"]
+"""File extension/output format."""
+
+type Column = Literal[
+    "date",
+    "time",
+    "delay",
+    "distance",
+    "origin",
+    "destination",
+    "ScheduledFlightDate",
+    "ScheduledFlightTime",
+    "DepDelay",
+]
+"""
+Columns available for ``flights`` datasets.
+
+Descriptions
+------------
+*date*
+    Either a **datetime-typed** value, or a formatted datetime string
+*time*
+    Either a **time-typed** value, or decimal hours.minutes when using decimal format (e.g., 6.5 for 6:30)
+*distance*
+    Flight distance in miles (integer)
+*delay*
+    Arrival delay in minutes (integer)
+*origin*
+    Origin airport code
+*destination*
+    Destination airport code
+*ScheduledFlightDate*
+    Original scheduled flight date (YYYY-MM-DD)
+*ScheduledFlightTime*
+    Original scheduled departure time (HHMM)
+*DepDelay*
+    Departure delay in minutes (integer)
+
+See Also
+--------
+- https://www.bts.gov/topics/airlines-and-airports/world-airport-codes
+- https://www.transtats.bts.gov/TableInfo.asp?gnoyr_VQ=FGJ&QO_fu146_anzr=b0-gvzr&V0s1_b0yB=D
+"""
+
+type YearMonthDay = tuple[int, int, int] | Sequence[int]
+"""Arguments passed to ``datetime.date(...)``."""
+
+type IntoDate = dt.date | dt.datetime | YearMonthDay
+"""Anything that can be converted into a ``datetime.date``."""
+
+type IntoDateRange = (
+    tuple[IntoDate, IntoDate] | Mapping[Literal["start", "end"], IntoDate]
+)
+"""Anything that can be converted into a ``DateRange``."""
+
+
+type PlScanCsv = (
+    str
+    | Path
+    | IO[str]
+    | IO[bytes]
+    | bytes
+    | list[str]
+    | list[Path]
+    | list[IO[str]]
+    | list[IO[bytes]]
+    | list[bytes]
+)
+
+
+class DateTimeFormat(StrEnum):
+    ISO = "iso"
+    DECIMAL = "decimal"
+    ACTUALLY_ISO = "ISO"
+
+
+DTF_TO_FMT: Mapping[DateTimeFormat, LiteralString] = {
+    DateTimeFormat.ISO: "%Y/%m/%d %H:%M",
+    DateTimeFormat.ACTUALLY_ISO: "iso",
+}
```

Comment:
### TODO
- [ ] Refactor `DateTimeFormat`, `DTF_TO_FMT` and [`_transform_temporal`](https://github.com/vega/vega-datasets/blob/4a05b5142b9c053e8cbb76813330cfa0c132fcab/scripts/flights2.py#L700)
	- Should not require an `Enum`, global mapping and a function
- [ ] Write doc(s) for whatever the final api is

Reply:
Resolved in (https://github.com/vega/vega-datasets/pull/645/commits/3418c5ada5e08357d628d120eb943c9d7bd012ae)